src/realm/object-store/c_api/query.cpp

```diff
@@ -231,9 +231,8 @@ RLM_API realm_query_t* realm_query_append_query(const realm_query_t* existing_qu
         auto realm = existing_query->weak_realm.lock();
         auto table = existing_query->query.get_table();
         Query query = parse_and_apply_query(realm, table, query_string, num_args, args);
-        Query combined = Query(existing_query->query).and_query(query);
+        Query combined = existing_query->query.and_query(query);
```

Comment:
What about the ordering? Which ordering should we consider?

Reply:
Last ordering should win, in my opinion.